sound/soc/codecs/hdac_hda.c

```diff
@@ -135,10 +171,11 @@ static int hdac_hda_dai_set_tdm_slot(struct snd_soc_dai *dai,
 
 	hda_pvt = snd_soc_component_get_drvdata(component);
 	pcm = &hda_pvt->pcm[dai->id];
+
 	if (tx_mask)
-		pcm[dai->id].stream_tag[SNDRV_PCM_STREAM_PLAYBACK] = tx_mask;
+		pcm->stream_tag[SNDRV_PCM_STREAM_PLAYBACK] = tx_mask;
```

Comment:
This was actually a bug in original code, but it happened to work by accident. Pcm is already dereferenced by dai->id on line 137/173. Considering this fix is only needed when we add HDMI/DP support, I included this in the same patch.